docs/rfc/rfc-5.md

```diff
@@ -0,0 +1,271 @@
+
+The Ethereum sidecar should be enhanced with an ability to fetch recent `AssetsUnlocked`entries submitted on the
+Mezo chain and attest them on Ethereum. The specific logic described below is supposed to be executed only by
+validators having the "bridge" privilege (the so called bridge validators). List of those validators can be
+determined by calling the `validatorsByPrivilege` method (with `1` as parameter) exposed by the `ValidatorPool`
+precompile.
+
+#### Monitoring loop
+
+The Ethereum sidecar should query its validator node privately to fetch recent `AssetsUnlocked` entries stored in
+the `x/bridge` module. The query mechanism is an implementation detail and is not enforced by this document. A
+possible approach is exposing `AssetsUnlocked` entries by the validator using EVM JSON-RPC or gRPC and periodically
+polling them from within the sidecar monitoring loop. What is important, the sidecar should not query any third-party
+node to avoid centralization and putting trust on external components.
+
+#### Attestation process
+
+Once `AssetsUnlocked`entries are fetched by the sidecar, all of them should be attested on Ethereum. The
+attestation process requires access to the operator private key used by the corresponding validator. This is necessary
+to attribute the given attestation to a specific bridge validator (as returned by the aforementioned
+`validatorsByPrivilege` method of the `ValidatorPool` precompile). The integration of the private key is an
+implementation detail but a possible solution may be using the keyring from within the sidecar process.
+
+The sidecar attestation process should look as follows:
+
+* Batch attestation phase:
+    * The sidecar produces an ECDSA signature over the given `AssetsUnlocked` entry, using the aforementioned operator
+      private key of the corresponding validator
+    * The sidecar submits the ECDSA signature to the Bridge Worker using an HTTP call. The Bridge Worker is an
+      off-chain component responsible for aggregating attestations and submitting them to the `MezoBridge` contract
+      in one batch.
+    * The sidecar monitors the progress of batch attestation for a specific time, defined by a new
+      `batch_attestation_timeout` parameter
+    * If the given `AssetsUnlocked`entry was properly attested by the Bridge Worker, the sidecar completes the
+      attestation process for this entry. Otherwise, the sidecar falls back to individual attestation.
+* Individual attestation phase:
+    * The sidecar submits an attestation by issuing an Ethereum transaction against the `MezoBridge` contract directly.
+      This step uses the operator private key to set the `msg.sender` of the issued transaction and does not require
+      attaching the aforementioned ECDSA signature.
+    * The sidecar confirms finality of the above Ethereum transaction and completes the attestation process for the
+      given `AssetsUnlocked`entry
```

Comment:
We should make it more explicit that the validators move automatically from the batch attestation phase to individual attestation phase if the batch attestation could not be performed.